packages/stryker/src/ScoreResultCalculator.ts

```diff
@@ -0,0 +1,125 @@
+        .sort(this.compareScoreResults);
+    } else {
+      return [];
+    }
+  }
+
+  private static determineCommonBasePath(results: MutantResult[], basePath: string) {
+    const uniqueFiles = _.uniq(results.map(result => result.sourceFilePath));
+    const uniqueFileDirectories = uniqueFiles.map(file => file.substr(basePath.length).split(path.sep));
+    if (uniqueFileDirectories.length) {
+      return uniqueFileDirectories
+        .reduce((previousDirectories, currentDirectories) => previousDirectories.filter((token, index) => currentDirectories[index] === token))
+        .join(path.sep);
+    } else {
+      return '/';
+    }
+  }
+
+  private static calculateNumbers(mutantResults: MutantResult[], childScores: ScoreResult[]) {
+    if (childScores.length) {
+      return this.calculateNumbersBasedOnChildScores(childScores);
+    } else {
+      return this.countNumbers(mutantResults);
+    }
+  }
+
+  private static calculateNumbersBasedOnChildScores(childScores: ScoreResult[]) {
+    const sum = (expression: (score: ScoreResult) => number) => _.sum(childScores.map(expression));
+    const totalDetected = sum(_ => _.totalDetected);
+    const totalUndetected = sum(_ => _.totalUndetected);
+    const totalMutants = sum(_ => _.totalMutants);
+    const totalCovered = sum(_ => _.totalCovered);
+    return {
+      killed: sum(_ => _.killed),
+      timedOut: sum(_ => _.timedOut),
+      survived: sum(_ => _.survived),
+      noCoverage: sum(_ => _.noCoverage),
+      errors: sum(_ => _.errors),
+      totalDetected,
+      totalUndetected,
+      totalMutants,
+      totalCovered,
+      mutationScore: totalDetected / totalMutants * 100 || 0,
+      mutationScoreBasedOnCoveredCode: totalDetected / totalCovered * 100 || 0
+    };
+  }
+
+  private static countNumbers(mutantResults: MutantResult[]) {
+    let killed = 0;
+    let timedOut = 0;
+    let survived = 0;
+    let noCoverage = 0;
+    let errors = 0;
+    mutantResults.forEach(mutation => {
+      switch (mutation.status) {
+        case MutantStatus.Killed:
+          killed++;
+          break;
+        case MutantStatus.TimedOut:
+          timedOut++;
+          break;
+        case MutantStatus.Survived:
+          survived++;
+          break;
+        case MutantStatus.NoCoverage:
+          noCoverage++;
+          break;
+        case MutantStatus.Error:
+          errors++;
+          break;
+      }
+    });
+    const totalDetected = timedOut + killed;
+    const totalUndetected = survived + noCoverage;
+    const totalMutants = totalDetected + totalUndetected;
+    const totalCovered = totalDetected + survived;
+    return {
+      killed,
+      survived,
+      noCoverage,
+      errors,
+      timedOut,
+      totalDetected,
+      totalUndetected,
+      totalMutants,
+      totalCovered,
+      mutationScore: Math.floor(totalDetected / totalMutants * 100) || 0,
```

Comment:
True. This is a major bug. We don't want to round any numbers here, if a reporter wants to do that, it can, but stryker should only give you the raw value